backend/inspirehep/pidstore/cli.py

```diff
@@ -138,3 +138,50 @@ def mint_bais(yes_i_know, create):
             "There are still external BAIs left in DB. Probably they are not properly assigned to authors."
             " Please fix them and re-run command!"
         )
+
+
+@inspire_pidstore.command("fast-mint-new-bais")
+@click.option("--yes-i-know", is_flag=True)
+@with_appcontext
+def fast_mint_bais(yes_i_know):
+    """Mint BAIs"""
+    if not current_app.config.get("FEATURE_FLAG_ENABLE_BAI_PROVIDER"):
+        click.echo(
+            "Cannot mint BAIs without FEATURE_FLAG_ENABLE_BAI_PROVIDER set!", err=True
+        )
+        return
+    if current_app.config.get("FEATURE_FLAG_ENABLE_AUTHOR_DISAMBIGUATION"):
+        click.echo(
+            "Cannot mint BAIs with FEATURE_FLAG_ENABLE_AUTHOR_DISAMBIGUATION set!",
+            err=True,
+        )
+    if not yes_i_know:
+        click.confirm(
+            "Do you want to create new BAIs for authors records which are missing one?",
+            abort=True,
+        )
+    current_app.config["FEATURE_FLAG_ENABLE_BAI_CREATION"] = True
+    all_authors = PersistentIdentifier.query.filter_by(
+        pid_type="aut", status="R"
+    ).with_entities("object_uuid")
+    bai_authors = PersistentIdentifier.query.filter_by(
+        pid_type="bai", status="R"
+    ).with_entities("object_uuid")
+    all_authors = {author[0] for author in all_authors}
+    bai_authors = {author[0] for author in bai_authors}
+    missing_bais = all_authors - bai_authors
+    length = len(missing_bais)
+    with click.progressbar(all_authors, length=length) as authors_bar:
+        for author_uuid in authors_bar:
+            author_record = AuthorsRecord.get_record(str(author_uuid))
+            author_data = dict(author_record)
+            BAIMinter.mint(author_uuid, author_data)
+            if author_data.get("ids") != author_record.get("ids"):
```

Comment:
I'm not sure this check is needed, as we know they have no BAI yet, and we know this is going to create one.

Reply:
Actually it's quite helpful as with this we are checking if BAI was actually created or not, as if there were no BAI and now there will be one it will differ. Otherwise if BAI was not created because of something then we do not want to update record. Additionally I can add a log to display info that BAI for this author was not created.